src/olympia/devhub/tasks.py

```diff
@@ -821,6 +821,32 @@ def resize_preview(src, preview_pk, **kw):
         log.error("Error saving preview: %s" % e)
 
 
+@task
+def recreate_previews(addon_pk, **kw):
+    log.info('[1@None] Recreating previews for addon: %s' % addon_pk)
+    addon = Addon.objects.get(pk=addon_pk)
+    for preview in addon.previews.all():
+        log.info('[1@None] Resizing preview: %s' % preview.id)
+        try:
+            if not preview.sizes:
+                preview.sizes = {}
+            if storage.exists(preview.original_path):
+                # We have an original size image, so we can resize that.
+                src = preview.original_path
+                (preview.sizes['image'], _) = resize_image(
+                    src, preview.image_path, amo.ADDON_PREVIEW_SIZES['full'])
+            else:
+                # Otherwise we can't create a new sized full image, but can
+                # use it for a new thumbnail
+                src = preview.image_path
+            # But we should resize the thumbnail regardless.
+            (preview.sizes['thumbnail'], _) = resize_image(
+                src, preview.thumbnail_path, amo.ADDON_PREVIEW_SIZES['thumb'])
+            preview.save()
+        except Exception as e:
+            log.error("Error saving preview: %s" % e)
```

Comment:
Either use `log.exception` or set `exc_info=True` so that we get the proper stack-trace of what happened. Also, this is very generic, I'd rather have a list of things that we know can go wrong and report everything else directly to sentry.

Reply:
I'd imagine exceptions would fall into the rough buckets of "full size preview doesn't exist" and "full/original size preview is an invalid image somehow" but I'm not sure what we'd do with that information other than shrug and log.  I suppose we could/should delete the `Preview` instance if that's happening?

Reply:
changed to log.exception in https://github.com/mozilla/addons-server/pull/8553/commits/d002da8dc8c33a60929d9ab9fec9643d928c28c2 but didn't delete the `Preview` instance.